modules/i3s/src/lib/parsers/parse-slpk/slpk-archieve.ts

```diff
@@ -0,0 +1,58 @@
+import {processOnWorker} from '@loaders.gl/worker-utils';
+import md5 from 'md5';
+import {CompressionWorker} from '@loaders.gl/compression';
+import LocalFileHeader from './local-file-header';
+
+export default class SlpkArchieve {
+  slpkArchieve: DataView;
+  hashArray: {hash: Buffer; offset: number}[];
+  constructor(slpkArchieveBuffer: ArrayBuffer, hashFile: ArrayBuffer) {
+    const hashFileBuffer = Buffer.from(hashFile);
+    this.slpkArchieve = new DataView(slpkArchieveBuffer);
+    this.hashArray = [];
+    for (let i = 0; i < hashFileBuffer.buffer.byteLength; i = i + 24) {
+      const offsetBuffer = new DataView(
+        hashFileBuffer.buffer.slice(
+          hashFileBuffer.byteOffset + i + 16,
+          hashFileBuffer.byteOffset + i + 24
+        )
+      );
+      const offset = offsetBuffer.getUint32(offsetBuffer.byteOffset, true);
+      this.hashArray.push({
+        hash: Buffer.from(
+          hashFileBuffer.subarray(hashFileBuffer.byteOffset + i, hashFileBuffer.byteOffset + i + 16)
+        ),
+        offset
+      });
+    }
```

Comment:
Avoid difficult logic in the constructor. Make `private initializeHashArray` method